.github/workflows/create-release.yml

```diff
@@ -0,0 +1,98 @@
+---
+name: Create GitHub release
+
+on:
+  push:
+    tags:
+      - '*v[0-9]*'
+
+jobs:
+  build-release-assets:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-14, macos-12]
+        arch: [amd64, arm64]
+        python-version: ['3.9', '3.10', '3.11', '3.12']
+        exclude:
+          - os: windows-latest
+            arch: arm64  # Windows doesn't support arm64 in this context
+          - os: macos-12
+            arch: amd64  # macOS 11 should be used for arm64 (Apple Silicon)
+          - os: macos-14
+            arch: amd64  # macOS 11 should be used for arm64 (Apple Silicon)
```

Comment:
I am a bit confused by this, the versions do not seem to match in the comments and code. Another thing is that it seems that it will build for arm64 macOS twice - on version 12 and 14, but not even once for amd64.